### PR TITLE
Add optical core track view and launcher

### PR DESCRIPTION
**Pull request by sethrj**
This defines optical kernel launchers, track view, and executor. It uses these to implement a boundary crossing kernel for optical tracks.

### CODE REVIEW

File: src/celeritas/optical/CoreTrackView.hh [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+    // Return a material view
+    inline CELER_FUNCTION MaterialView material() const;
+
+    // Return a material view (using an existing geo view
+    inline CELER_FUNCTION MaterialView material(GeoTrackView const&) const;
+
+    // Return a simulation management view
+    inline CELER_FUNCTION SimTrackView sim() const;
+
+    // Return a particle view
```
Comment (amandalund) [inline src/celeritas/optical/CoreTrackView.hh]:
```suggestion
    // Return a physics view
```

File: src/celeritas/optical/TrackView.hh [hunk 2]
```diff
@@ -19,6 +19,7 @@ namespace optical
  * Temporary class for testing optical interactors.
  *
  * \todo Fill with actual code.
+ * \todo Rename ParticleTrackView (?) Should polarization live here as well?
```
Comment (amandalund) [inline src/celeritas/optical/TrackView.hh:22]: Sure, makes sense.

File: src/celeritas/optical/CoreTrackView.hh [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    //!@}
+
+  public:
+    // Construct directly from a track slot ID
+    inline CELER_FUNCTION CoreTrackView(ParamsRef const& params,
+                                        StateRef const& states,
+                                        TrackSlotId slot);
+
+    // Return a geometry view
+    inline CELER_FUNCTION GeoTrackView geometry() const;
```
Comment (amandalund) [inline src/celeritas/optical/CoreTrackView.hh:42]: The shorter names are nice... were you thinking of eventually updating the main `CoreTrackView` to match?

**Review by amandalund — APPROVED** [review]
Awesome, thanks @sethrj!

File: src/celeritas/optical/CoreTrackView.hh (hunk 3 above)
Comment (sethrj) [reply src/celeritas/optical/CoreTrackView.hh:42]: Yep indeed.

**Comment by sethrj** [comment]
Ugh I'm an idiot. I thought all the tests passed, but because I'd marked as draft the docker ones didn't. Going to fix this now...

**Pull request merged**